runtime/array.c

```diff
@@ -327,43 +315,12 @@ CAMLprim value caml_floatarray_blit(value a1, value ofs1, value a2, value ofs2,
 CAMLprim value caml_array_blit(value a1, value ofs1, value a2, value ofs2,
                                value n)
 {
-  value * src, * dst;
-  intnat count;
-
 #ifdef FLAT_FLOAT_ARRAY
   if (Tag_val(a2) == Double_array_tag)
     return caml_floatarray_blit(a1, ofs1, a2, ofs2, n);
```

Comment:
Is it safe to use `caml_flotarray_blit` here? The same atomicity concerns that prevents us from using `memmove` and `memcpy` for non-float arrays applies here, doesn't it? See atomicity comment in `caml_blit_fields` https://github.com/ocaml/ocaml/pull/10831/files#diff-619cc3868df8ee59cc68190a3cb39c9403184abe22688543ede23506b2f781a0R338-R339. If the `memmove` copies fields by smaller than word size, we may see a floating-point value read from the array which was not written by the program.

Reply:
Fixed by https://github.com/ocaml-multicore/ocaml-multicore/pull/822.